File: Data Structures & Algorithms/buy-and-sell-crypto/submission-2.py
class Solution:
    def maxProfit(self, prices):
        profit = 0

        if len(prices) <= 1 : return profit

        left = 0 
        right = 1

        while right < len(prices):
            if prices[left] < prices[right]:
                profit = max(profit, prices[right] - prices[left])
            else:
                left = right
            right += 1

        return profit

File: Data Structures & Algorithms/grumpy-bookstore-owner/submission-2.java
class Solution {
    public int maxSatisfied(int[] customers, int[] grumpy, int minutes) {
        int max = 0;
        
        for(int i = 0; i < customers.length; i++){
            if(grumpy[i] == 0) max += customers[i];
        }

        if(minutes == 0) return max; 

        int left = 0;
        int right = 0;
        int current = max;
        while(right  < customers.length){
            if(right - left < minutes || left >= right){
                
                if(grumpy[right] == 1){
                    current += customers[right];
                }
                right++;
                max = Math.max(max, current);
            } else {
                if(grumpy[left] == 1){
                    current -= customers[left];
                }
                left++;
                max = Math.max(max, current);
            }
        }

        return max;
    }
}